modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/FrontendDataSetContext.tsx

```diff
@@ -35,6 +35,7 @@ export interface IFrontendDataSetContext {
 	nestedItemsReferenceKey?: string;
 	onActionDropdownItemClick: Function;
 	onBulkActionItemClick: Function;
+	onSearch: ({query}: {query: string}) => void;
```

Comment:
Refactoring `updateSearchParam` to callback pattern, to have better separation of concerns.